diff --git a/lib/libUPnP/Neptune/Source/Core/NptHttp.cpp b/lib/libUPnP/Neptune/Source/Core/NptHttp.cpp
index 07586f9..158ab90 100644
--- a/lib/libUPnP/Neptune/Source/Core/NptHttp.cpp
+++ b/lib/libUPnP/Neptune/Source/Core/NptHttp.cpp
@@ -2880,14 +2880,14 @@ NPT_HttpFileRequestHandler_DefaultFileTypeMap[] = {
     {"aif",  "audio/x-aiff"},
     {"aifc", "audio/x-aiff"},
     {"aiff", "audio/x-aiff"},
-    {"mka",  "audio/x-matroska"},
+    {"mka",  "audio/x-mka"},
     {"mpa",  "audio/mpeg"},
     {"mp2",  "audio/mpeg"},
     {"mp3",  "audio/mpeg"},
     {"m4a",  "audio/mp4"},
     {"wma",  "audio/x-ms-wma"},
     {"wav",  "audio/x-wav"},
-    {"mkv",  "video/x-matroska"},
+    {"mkv",  "video/x-mkv"},
     {"mpeg", "video/mpeg"},
     {"mpg",  "video/mpeg"},
     {"mp4",  "video/mp4"},
@@ -2899,7 +2899,6 @@ NPT_HttpFileRequestHandler_DefaultFileTypeMap[] = {
     {"wmv",  "video/x-ms-wmv"},
     {"wtv",  "video/x-ms-wmv"},
     {"asf",  "video/x-ms-asf"},
-    {"mkv",  "video/x-matroska"},
     {"mk3d", "video/x-matroska-3d"},
     {"flv",  "video/x-flv"},
     {"avi",  "video/x-msvideo"},
diff --git a/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp b/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp
index 562315d..c288a1e 100644
--- a/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp
+++ b/lib/libUPnP/Platinum/Source/Core/PltProtocolInfo.cpp
@@ -97,8 +97,9 @@ PLT_HttpFileRequestHandler_DefaultDlnaMap[] = {
     {"video/x-ms-wmv", "DLNA.ORG_PN=WMVHIGH_BASE;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000"},
     {"video/x-msvideo","DLNA.ORG_PN=AVI;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000"},
     {"video/x-ms-asf", "DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000"},
+    {"video/x-mkv", "DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000"},
     {"video/x-matroska","DLNA.ORG_OP=01;DLNA.ORG_CI=0"},
-    {"video/x-flv",    "DLNA.ORG_OP=01;DLNA.ORG_CI=0"},
+    {"video/x-flv",    "DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01500000000000000000000000000000"},
 };
 
 static const PLT_HttpFileRequestHandler_DefaultDlnaExtMapEntry 
diff --git a/xbmc/utils/Mime.cpp b/xbmc/utils/Mime.cpp
index 266b9a5..e3ecf6d 100644
--- a/xbmc/utils/Mime.cpp
+++ b/xbmc/utils/Mime.cpp
@@ -235,8 +235,8 @@ map<string, string> fillMimeTypes()
   mimeTypes.insert(pair<string, string>("mime",      "message/rfc822"));
   mimeTypes.insert(pair<string, string>("mjf",       "audio/x-vnd.audioexplosion.mjuicemediafile"));
   mimeTypes.insert(pair<string, string>("mjpg",      "video/x-motion-jpeg"));
-  mimeTypes.insert(pair<string, string>("mka",       "audio/x-matroska"));
-  mimeTypes.insert(pair<string, string>("mkv",       "video/x-matroska"));
+  mimeTypes.insert(pair<string, string>("mka",       "audio/x-mka"));
+  mimeTypes.insert(pair<string, string>("mkv",       "video/x-mkv"));
   mimeTypes.insert(pair<string, string>("mm",        "application/x-meme"));
   mimeTypes.insert(pair<string, string>("mme",       "application/base64"));
   mimeTypes.insert(pair<string, string>("mod",       "audio/mod"));
